code/modules/shuttle/docking.dm

```diff
@@ -173,6 +173,16 @@
 		var/turf/oldT = old_turfs[i]
 		var/turf/newT = new_turfs[i]
 		newT.afterShuttleMove(oldT, rotation) //turfs
+		var/turf/newCeiling = get_step_multiz(newT, UP)
+		if(newCeiling)
+			if(istype(newCeiling, /turf/open/openspace))
+				newCeiling.ChangeTurf(/turf/open/floor/engine/hull/ceiling, list(/turf/open/openspace))
+			else if (istype(newCeiling, /turf/open/space/openspace))
+				newCeiling.ChangeTurf(/turf/open/floor/engine/hull/ceiling, list(/turf/open/space/openspace))
+		var/turf/oldCeiling = get_step_multiz(oldT, UP)
+		if(oldCeiling && istype(oldCeiling, /turf/open/floor/engine/hull/ceiling))
+			var/turf/open/floor/engine/hull/ceiling/oldShuttleCeiling = oldCeiling
+			oldShuttleCeiling.ChangeTurf(oldShuttleCeiling.old_turf_type)
```

Comment:
Can you change these local vars to use camel_case? Also, this block of code is kinda dense, comments explaining its purpose would be real handy.